docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
+through the [CRI-O](https://github.com/kubernetes-incubator/cri-o)
+and [Containerd](https://github.com/containerd/containerd)
+implementations.
+
+Kata Containers provides a ["shimv2"](#shim-v2-architecture) compatible runtime.
+
+## Shim v2 architecture
+
+The Kata Containers runtime is shim v2 ("shimv2") compatible. This
+section explains what this mean.
+
+### History
+
+In the old [Kata 1.x architecture](https://github.com/kata-containers/documentation/blob/master/design/architecture.md),
+the Kata [runtime](#runtime) was an executable called `kata-runtime`.
+The container manager called this executable multiple times when
+creating each container. Each time the runtime was called a different
+OCI command-line verb was provided. This architecture was simple, but
+not well suited to creating VM based containers due to the issue of
+handling state between calls. Additionally, the architecture suffered
+from performance issues related to continually having to spawn new
+instances of the runtime binary and the [Kata shim](https://github.com/kata-containers/shim)) and Kata
+proxy](https://github.com/kata-containers/proxy) processes for systems
+that did not provide VSOCK.
+
+### An improved architecture
+
+The
+[containerd runtime shimv2 architecture](https://github.com/containerd/containerd/tree/master/runtime/v2)
+or "shim API" architecture resolves the issues with the old
+architecture by defining as set of shimv2 APIs that a compatible
+runtime implementation must supply. Rather than calling the runtime
+binary multiple times for each new container, the shimv2 architecture
+runs a single instance of the runtime binary (for any number of
+containers). This improves performance and resolves the state handling
+issue.
+
+The shimv2 API is similar to the
+[OCI runtime](https://github.com/opencontainers/runtime-spec)
+API in terms of the way the container lifecycle is split into
+different verbs. Rather than calling the runtime multiple times, the
+container manager creates a socket and passes it to the shimv2
+runtime. The socket is a bi-directional communication channel that
+uses a gRPC based protocol to allow the container manager to send API
+calls to the runtime, which returns the result to the container
+manager using the same channel.
+
+The shimv2 architecture allows running several containers per VM to
+support container engines that require multiple containers running
+inside a pod.
+
+With the new architecture [Kubernetes](#kubernetes-support)) can
+launch Pod and OCI compatible containers with a single runtime shim
+(`containerd-shim-kata-v2`) per Pod instead of `2N+1` shims. No stand alone
+`kata-proxy` process is required, even if VSOCK is not available.
+
+### Architectural comparison
+
+| Kata version | Kata Runtime process calls | Kata shim processes | Kata proxy processes (if no VSOCK) |
+|-|-|-|-|
+| 1.x | multiple / container | 1 / container connection | 1 |
+| 2.x | 1 / VM | 0 | 0 |
```

Comment:
This is kind of a nit, but it looks the numbers are being divided.   I'd consider saying  "multiple per container"   "1 per VM"  Could you make the comparison "Multiple per VM" and "One per VM" instead?  My mind starts to wonder what the difference is between container and VM in this scenario.

Reply:
Looks better.